### ISSUE TITLE
openrc fails to build with musl-gcc

### ISSUE THREAD

**Issue by austin987** [OpenRC/openrc#259]
Noticed because of #144

If I enable musl-gcc in TravisCI, it fails to build. Talking to @williamh, this is expected to work. Current failure:
```
===> src/libeinfo (all)
make[2]: Entering directory `/home/travis/build/austin987/openrc/src/libeinfo'
rm -f .depend
musl-gcc -I../includes -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L  -MM libeinfo.c > .depend
make[2]: Leaving directory `/home/travis/build/austin987/openrc/src/libeinfo'
make[2]: Entering directory `/home/travis/build/austin987/openrc/src/libeinfo'
musl-gcc  -I../includes -O2 -g -std=c99 -Wall -Wextra -Wimplicit -Wshadow -Wformat=2 -Wmissing-prototypes -Wmissing-declarations -Wmissing-noreturn -Wmissing-format-attribute -Wnested-externs -Winline -Wwrite-strings -Wcast-align -Wcast-qual -Wpointer-arith -Wdeclaration-after-statement -Wsequence-point -Werror=implicit-function-declaration   -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L  -c libeinfo.c -o libeinfo.o
libeinfo.c: In function â€˜colour_terminalâ€™:
libeinfo.c:466:4: error: implicit declaration of function â€˜strlcpyâ€™ [-Werror=implicit-function-declaration]
    strlcpy(tmp, tgoto(bold, 0, 0), sizeof(tmp));
    ^
libeinfo.c:466:4: warning: nested extern declaration of â€˜strlcpyâ€™ [-Wnested-externs]
libeinfo.c:467:4: error: implicit declaration of function â€˜strlcatâ€™ [-Werror=implicit-function-declaration]
    strlcat(tmp, tgoto(_af, 0, c & 0x07), sizeof(tmp));
    ^
libeinfo.c:467:4: warning: nested extern declaration of â€˜strlcatâ€™ [-Wnested-externs]
libeinfo.c: In function â€˜elogvâ€™:
libeinfo.c:525:3: error: implicit declaration of function â€˜vsyslogâ€™ [-Werror=implicit-function-declaration]
   vsyslog(level, fmt, apc);
   ^
libeinfo.c:525:3: warning: nested extern declaration of â€˜vsyslogâ€™ [-Wnested-externs]
cc1: some warnings being treated as errors
make[2]: *** [libeinfo.o] Error 1
musl-gcc -fPIC -DPIC  -I../includes -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L  -O2 -g -std=c99 -Wall -Wextra -Wimplicit -Wshadow -Wformat=2 -Wmissing-prototypes -Wmissing-declarations -Wmissing-noreturn -Wmissing-format-attribute -Wnested-externs -Winline -Wwrite-strings -Wcast-align -Wcast-qual -Wpointer-arith -Wdeclaration-after-statement -Wsequence-point -Werror=implicit-function-declaration   -c libeinfo.c -o libeinfo.So
libeinfo.c: In function â€˜colour_terminalâ€™:
libeinfo.c:466:4: error: implicit declaration of function â€˜strlcpyâ€™ [-Werror=implicit-function-declaration]
    strlcpy(tmp, tgoto(bold, 0, 0), sizeof(tmp));
    ^
libeinfo.c:466:4: warning: nested extern declaration of â€˜strlcpyâ€™ [-Wnested-externs]
libeinfo.c:467:4: error: implicit declaration of function â€˜strlcatâ€™ [-Werror=implicit-function-declaration]
    strlcat(tmp, tgoto(_af, 0, c & 0x07), sizeof(tmp));
    ^
libeinfo.c:467:4: warning: nested extern declaration of â€˜strlcatâ€™ [-Wnested-externs]
libeinfo.c: In function â€˜elogvâ€™:
libeinfo.c:525:3: error: implicit declaration of function â€˜vsyslogâ€™ [-Werror=implicit-function-declaration]
   vsyslog(level, fmt, apc);
   ^
libeinfo.c:525:3: warning: nested extern declaration of â€˜vsyslogâ€™ [-Wnested-externs]
cc1: some warnings being treated as errors
```

Reporting so someone who cares can fix it. If it's not supported, feel free to close this.

**Comment by concatime** [comment]
For `strlcpy`, https://wiki.musl-libc.org/faq.html#Q:-I'm-trying-to-compile-something-against-musl-and-I-get-lots-of-error-messages!

**Comment by austin987** [comment]
Thanks for the tip @concatime. I sent https://github.com/OpenRC/openrc/pull/261 to fix this.

**Comment by williamh** [comment]
This is fixed in 0.41.x.
